libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -1112,8 +1100,10 @@ blk_hash_t PbftManager::calculateOrderHash(std::vector<DagBlock> const &dag_bloc
 
 std::pair<blk_hash_t, bool> PbftManager::proposeMyPbftBlock_() {
   auto round = getPbftRound();
-  // In propose block, only use first weighted index 0 for VRF sortition
-  if (weighted_votes_count_ == 0 || !shouldSpeak(propose_vote_type, round, step_, 0)) {
+  // In propose block, only use one vote for VRF sortition
+  VrfPbftSortition vrf_sortition(vrf_sk_, {propose_vote_type, propose_vote_type, step_});
+  if (weighted_votes_count_ == 0 ||
+      !vrf_sortition.getBinominalDistribution(1, sortition_threshold_, getDposTotalVotesCount())) {
```

Comment:
Here is wrong. In generateVote you don't only use stake 1 to generate proposal vote. Here need match. Either:
1. For any proposal_vote_type, only use stake 1 to generate vote
2. Or change here, using real stake to generate block

Reply:
And same here, should switch sortition_threshold_ and getDposTotalVotesCount()?